webpack/assets/javascripts/react_app/common/globalIdHelpers.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Helpers for dealing with GlobalIDs as returned by GraphQL. When `id` property is
+ * retrieved through GraphQL, the value is not a "native" ID under which the
+ * backend knows the record in question.
+ *
+ * The GlobalID contains a version number, model class and the id itself, put
+ * together into a single string and encoded as base64.
+ */
+
+const idSeparator = '-';
+const versionSeparator = ':';
+const defaultVersion = '01';
+
+// Decodes the provided GlobalID, returns a three element array consisting of
+// version, type name and id as strings
+export const decode = globalId => {
+  const s = Buffer.from(globalId, 'base64').toString();
+  let i = s.indexOf(versionSeparator);
+  const version = parseInt(s.substring(0, i), 10);
+  const payload = s.substring(i + 1);
+
+  i = payload.indexOf(idSeparator);
+  const typeName = payload.substring(0, i);
+  const id = payload.substring(i + 1);
+
+  return [version, typeName, id];
```

Comment:
```suggestion
  return {version, typeName, id};
```
this should be an object as every item has a different meaning

Reply:
@adamruzicka WDYT about ^^ I believe it would be bit more usable as object, then a sorted array, OTH we do have it in array on backend, so having it in object would break the consistency, right?

Reply:
Personally I'd prefer a 3-tuple and array was the next closest thing. If it being an object would be considered idiomatic js then I don't mind. I'd say in its decoded and broken-up-to-parts state it is more of an intermediate representation rather than something that would have to exactly match how its used in backend's internals.

Reply:
object would definitelly be better fit within JS idiomatics and if `decode` is supposed to be used by anything outside this file, I think we would better return an object, but if that would only be understood as intermediate state, then I'd not export the `decode` function and only `getId` functions etc. and in such case array is fine.